File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    levelOrder(root) {
        const res = [];
        let queue = [];
        if (root){
            queue.push(root);
        }
        while (queue.length){
            let tempArr = [];

            let length = queue.length;
            for (let i = 0; i < length; i++){
                let curr = queue.shift();
                if(root!==null){
                    tempArr.push(curr.val);  
                }
                if (curr.left){
                    queue.push(curr.left);
                }
                if (curr.right){
                    queue.push(curr.right);
                }
            }
            res.push(tempArr);
        }

        return res;
    }
}

File: Data Structures & Algorithms/decode-ways/submission-2.py
class Solution:
    def numDecodings(self, s: str) -> int:
        def dfs(i : int) -> int:
            if i >= len(s):
                return 1
            elif s[i] == "0":
                return 0

            res = dfs(i + 1)
            if i + 1 < len(s) and int(s[i:i+2]) <= 26:
                res += dfs(i+2)
            return res    

        res = dfs(0)
        return res